types/react/test/hooks.tsx

```diff
@@ -201,6 +201,14 @@ function useEveryHook(ref: React.Ref<{ id: number }>|undefined): () => boolean {
     // make sure the generic argument does reject actual potentially undefined inputs
     // $ExpectError
     React.useState<number>(undefined)[0];
+    // When state is a function type, it must be returned from a function,
+    // not provided directly.
+    // $ExpectError
+    React.useState<(() => boolean)>(() => true)[1](() => false);
+    // Returning a function for state is fine
+    React.useState<(() => boolean)>(() => true)[1](() => () => false);
```

Comment:
The `() => true` initializer will create a `boolean` state (`useState` will treat function arguments as lazy initializers instead of values)